proof_system/tests/blind_bbs_plus_sig_and_message_equality.rs

```diff
@@ -0,0 +1,155 @@
+// cargo-fmt makes this much harder to read
+use ark_bls12_381::{Bls12_381, Fr};
+use ark_std::{
+    collections::{BTreeMap, BTreeSet},
+    rand::{prelude::StdRng, SeedableRng},
+    UniformRand,
+};
+use bbs_plus::prelude::SignatureG1;
+use blake2::Blake2b512;
+use proof_system::{
+    prelude::{EqualWitnesses, MetaStatements, Proof, ProofSystemError, WitnessRef, Witnesses},
+    proof_spec::ProofSpec,
+    statement::{
+        bbs_plus::{
+            PoKBBSSignatureG1Prover as PoKSignatureBBSG1ProverStmt,
+            PoKBBSSignatureG1Verifier as PoKSignatureBBSG1VerifierStmt,
+        },
+        Statements,
+    },
+    witness::PoKBBSSignatureG1 as PoKSignatureBBSG1Wit,
+};
+use test_utils::bbs::*;
+
+#[test]
+fn blind_bbs_plus_sig_and_message_equality() {
+    // This test
+    // - creates two issuers/signers
+    // - creates one signature for each signer, including a common attribute, blinded (by the holder) for each signature
+    // - it then unblinds the signatures and
+    // - creates a proof of knowledge of both signatures and equality of the blinded attributes
+    // It tests that
+    // - when the blinded values are   equal, proof creation and verification both succeeds
+    // - when the blinded values are unequal, either proof creation fails, or verification of the created proof fails
+    //
+    // This test was motivated by the fact that the existing blinding tests
+    //   proof_system/tests/bbs_plus_and_accumulators.rs
+    //     requesting_partially_blind_bbs_plus_sig
+    //     requesting_partially_blind_bbs_plus_sig
+    //     requesting_partially_blind_bbs_sig
+    // do not use the unblinded/blind signature in proofs.  Hence, this test.
+
+    blind_bbs_plus_sig_and_message_equality_aux(true).unwrap();
+    if blind_bbs_plus_sig_and_message_equality_aux(false).is_ok() {
```

Comment:
Any reason you wouldn't want `assert!(blind_bbs_plus_sig_and_message_equality_aux(false).is_err())` to replace this `if` block.

Reply:
No reason :smile:.  Fixed, thanks.